src/ansys_sphinx_theme/__init__.py

```diff
@@ -107,6 +109,106 @@ def setup_default_html_theme_options(app):
+        The document tree for the page.
+
+    Notes
+    -----
+    .. [1] Originally implemented by `Alex Kaszynski <https://github.com/akaszynski>`_ in
+    `PyVista <https://github.com/pyvista/pyvista>`_,
+    see https://github.com/pyvista/pyvista/pull/4113
+    """
+
+    def fix_edit_link_button(link: str) -> str:
+        """Transform "edit on GitHub" links to the correct URL.
+
+        This function fixes the URL for the "edit this page" link.
+
+        Parameters
+        ----------
+        link : str
+            The link to the GitHub edit interface.
+
+        Returns
+        -------
+        str
+            The link to the corresponding file on the main branch.
+        """
+        github_user = context.get("github_user", "")
+        github_repo = context.get("github_repo", "")
+        github_source = context.get("source_path", "")
+        kind = context.get("github_version", "")
+
+        if pagename.startswith("examples") and "index" not in pagename:
+            return f"http://github.com/{github_user}/{github_repo}/edit/{kind}/{pagename}.py"
+
+        elif "_autosummary" in pagename:
+            for obj_node in list(doctree.findall(addnodes.desc)):
+                domain = obj_node.get("domain")
+                for signode in obj_node:
+                    if not isinstance(signode, addnodes.desc_signature):
+                        continue
+                    # Convert signode to a specified format
+                    info = {}
+                    for key in DOMAIN_KEYS.get(domain, []):
+                        value = signode.get(key)
+                        if not value:
+                            value = ""
+                        info[key] = value
+                    if not info:
+                        continue
+                    # This is an API example
+                    return sphinx_linkcode_resolve(
+                        domain=domain,
+                        info=info,
+                        library=f"{github_user}/{github_repo}",
+                        source_path=github_source,
+                        github_version=kind,
+                        edit=True,
+                    )
+
+        elif "autoapi" in pagename:
+            for obj_node in list(doctree.findall(addnodes.desc)):
+                domain = obj_node.get("domain")
+                if domain != "py":
+                    return link
+
+                for signode in obj_node:
+                    if not isinstance(signode, addnodes.desc_signature):
+                        continue
+
+                    fullname = signode["module"]
+                    modname = fullname.replace(".", "/")
+
+                    if github_source:
+                        return f"http://github.com/{github_user}/{github_repo}/edit/{kind}/{github_source}/{modname}.py"  # noqa: E501
+                    else:
+                        return f"http://github.com/{github_user}/{github_repo}/edit/{kind}/{modname}.py"  # noqa: E501
+
```

Comment:
Also... apparently this is valid for C, C++, Python and JavaScript. But our extension is assuming Python files as source code... Can we do something about this? Automate it if possible.

Reply:
yeah thats true, i will adapt that , thanks @RobPasMue